ultravox/training/config_base.py

```diff
@@ -3,14 +3,32 @@
 import logging
 import os
 from pathlib import Path
-from typing import List, Optional
+from typing import Any, Dict, List, Optional
 
 import simple_parsing
 import torch
+from pydantic import BaseModel
 
 from ultravox.model import ultravox_config
 
 
+class DataDictConfig(BaseModel):
+    # Path to the dataset, or huggingface dataset id
+    path: str
+    # Name of the dataset, or huggingface dataset config/subset
+    name: Optional[str] = None
+    splits: List[str] = dataclasses.field(default_factory=list)
```

Comment:
Do we need to set up different splits for train and validation?

Reply:
If we don't use the generic dataset class, the training and validation datasets are specified in the config yaml file as follows:

```
data_sets: ["gigaspeech"]
val_sets: ["heysquad_human", "anyinstruct", "soda", "peoplespeech"]
```

With the generic dataset class, the additional training datasets can be specified by:

```
data_dicts:
  - path: "fixie-ai/librispeech_asr"
    name: "clean"
    splits:
      - "train.100"
      - "train.360"
    user_template: "Continue the following text using less than 50 words:\n\n<|audio|>"
    assistant_template: "{{ continuation }}"
    transcript_template: "{{ text }}"
    num_samples: 100_000
  - path: "fixie-ai/librispeech_asr"
    name: "other"
    splits:
      - "train.500"
    user_template: "Continue the following text using less than 50 words:\n\n<|audio|>"
    assistant_template: "{{ continuation }}"
    transcript_template: "{{ text }}"
    num_samples: 100_000
```

Similar to the use of `--data_sets`, the datasets (and their corresponding splits) specified in `--data_dicts` are used for training. We could also include a `--val_dicts` option to specify additional validation datasets in a similar manner, but this feature is not yet implemented.

I believe it would be better to use a YAML configuration to specify all the details for training, validation, and evaluation datasets (with defaults specified in the default config YAML). We can address this in a future PR, as it requires extensive testing to support all existing datasets.